Data types

arthur-gyulabyan opened the pull request:
Data types section translated.

arthur-gyulabyan committed:
feat: translate article.md of data types section

arthur-gyulabyan committed:
feat: translate task.md of data types section

arthur-gyulabyan committed:
feat: translate solution.md of data types section

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/javascript-tutorial/hy.javascript.info?pullRequest=77) <br/>All committers have signed the CLA.

bugron commented:
@Arthur-Gyulabyan thanks for the PR. I'll review it soon. Keep up the good work!

arthur-gyulabyan commented:
@bugron thank you for response.

arthur-gyulabyan committed:
feat: translate article.md of interaction: alert, prompt, confirm section

arthur-gyulabyan committed:
feat: translate task.md of interaction: alert, prompt, confirm section

arthur-gyulabyan committed:
feat: translate solution.md of interaction: alert, prompt, confirm section

bugron commented:
This looks great. I'll be reviewing your translations today. For future let's keep the one article - one PR principle.

arthur-gyulabyan commented:
Oh, sorry for that mistake. I'll follow the principle in further PRs.

bugron commented on 1-js/02-first-steps/05-types/article.md:
```diff
@@ -1,226 +1,250 @@
+JavaScript-ում արժեքը միշտ պատկանում է տվյալների որոշակի տիպի։ Օրինակ՝ տող (string) կամ թիվ (number)։
 
-There are eight basic data types in JavaScript. Here, we'll cover them in general and in the next chapters we'll talk about each of them in detail.
+JavaScript-ում կան տվյալների ութ հիմնական տիպեր։ Այս գլխում մենք կանրադառնանք դրանց մասնակիորեն, իսկ հաջորդ գլուխներում կխոսենք դրանցից 
```
No need for a line break after `կխոսենք դրանցից`.

bugron commented on 1-js/02-first-steps/05-types/article.md:
````diff
@@ -1,226 +1,250 @@
 ```
 
-The *number* type represents both integer and floating point numbers.
+*number* տիպը ներկայացնում է ինչպես ամբողջ թվերը, այնպես էլ լողացող կետով թվերը։
````
```suggestion
*number* տիպը ներկայացնում է ինչպես ամբողջ, այնպես էլ տասնորդական թվերը։
```
A better suiting term for `floating point number` in Armenian would be `տասնորդական թվեր`. What do you think?

bugron commented on 1-js/02-first-steps/05-types/article.md:
```diff
@@ -1,226 +1,250 @@
-Besides regular numbers, there are so-called "special numeric values" which also belong to this data type: `Infinity`, `-Infinity` and `NaN`.
+Բացի սովորական թվերից կան նաև այսպես կոչված «հատուկ թվային արժեքներ», որոնք նույնպես պատկանում են այս տվյալների տիպին․ `Infinity`, 
+`-Infinity` 
+և `NaN`։
```
Also no need for line breaks here. Let's keep them on the same line. This would also help us review merge conflicts when say English article is translated and we're syncing with upstream.

bugron commented on 1-js/02-first-steps/05-types/article.md:
```diff
@@ -1,226 +1,250 @@
+և `NaN`։
 
-- `Infinity` represents the mathematical [Infinity](https://en.wikipedia.org/wiki/Infinity) ∞. It is a special value that's greater than any number.
+- `Infinity`-ն ներկայացնում է մաթեմատիկական [անվերջությունը](https://en.wikipedia.org/wiki/Infinity) ∞։ Այն հատուկ արժեք է, որը մեծ է 
```
Let's use [Armenian wiki page](https://hy.wikipedia.org/wiki/%D4%B1%D5%B6%D5%BE%D5%A5%D6%80%D5%BB%D5%B8%D6%82%D5%A9%D5%B5%D5%B8%D6%82%D5%B6_(%D5%B4%D5%A1%D5%A9%D5%A5%D5%B4%D5%A1%D5%BF%D5%AB%D5%AF%D5%A1)). In this project it is recommended to use appropriate wiki page translations if they exist. Otherwise we go for English articles.

bugron commented on 1-js/02-first-steps/05-types/article.md:
```diff
@@ -1,226 +1,250 @@
 
-- `Infinity` represents the mathematical [Infinity](https://en.wikipedia.org/wiki/Infinity) ∞. It is a special value that's greater than any number.
+- `Infinity`-ն ներկայացնում է մաթեմատիկական [անվերջությունը](https://en.wikipedia.org/wiki/Infinity) ∞։ Այն հատուկ արժեք է, որը մեծ է 
+  ցանկացած թվից։
```
No need for `ցանկացած թվից։` to be on a new line.

bugron commented on 1-js/02-first-steps/05-types/article.md:
```diff
@@ -1,226 +1,250 @@
+  ցանկացած թվից։
 
-    We can get it as a result of division by zero:
+    Մենք կարող ենք ստանալ այն որպես զրոի վրա բաժանման արդյունք,
```
```suggestion
    Մենք կարող ենք ստանալ այն որպես զրոյի վրա բաժանման արդյունք,
```

bugron commented on 1-js/02-first-steps/05-types/article.md:
````diff
@@ -1,226 +1,250 @@
-```smart header="Mathematical operations are safe"
-Doing maths is "safe" in JavaScript. We can do anything: divide by zero, treat non-numeric strings as numbers, etc.
+```smart header="Մաթեմատիկական գործողությունները անվտանգ են"
+Մաթեմատիկական գործողությունները JavaScript-ում «անվտանգ» են։ Մենք կարող ենք կատարել ցանկացած գործողություն՝ բաժանել զրոի, ոչ թվային տողերի 
````
```suggestion
Մաթեմատիկական գործողությունները JavaScript-ում «անվտանգ» են։ Մենք կարող ենք կատարել ցանկացած գործողություն՝ բաժանել զրոյի, ոչ թվային տողերի 
```

bugron commented on 1-js/02-first-steps/05-types/article.md:
````diff
@@ -1,226 +1,250 @@
-Doing maths is "safe" in JavaScript. We can do anything: divide by zero, treat non-numeric strings as numbers, etc.
+```smart header="Մաթեմատիկական գործողությունները անվտանգ են"
+Մաթեմատիկական գործողությունները JavaScript-ում «անվտանգ» են։ Մենք կարող ենք կատարել ցանկացած գործողություն՝ բաժանել զրոի, ոչ թվային տողերի 
+հետ վարվել ինչպես թվերի հետ։
````
No need for a new line here.

bugron commented on 1-js/02-first-steps/05-types/article.md:
```diff
@@ -1,226 +1,250 @@
+որոնք մեծ են <code>(2<sup>53</sup>-1)</code>-ից (նույնն է, ինչ 
+`9007199254740991`), 
+կամ փոքր են <code>-(2<sup>53</sup>-1)</code>-ից՝ բացասական թվերի դեպքում։ Այս տեխիկական սահմանափակումը պայմանավորված է թվերի ներքին 
+ներկայացման յուրահատկություններով։
```
Those should be on a single line.

bugron commented on 1-js/02-first-steps/05-types/article.md:
```diff
@@ -1,226 +1,250 @@
+ներկայացման յուրահատկություններով։
 
-For most purposes that's quite enough, but sometimes we need really big numbers, e.g. for cryptography or microsecond-precision timestamps.
+Հիմնական դեպքերում սա ավելին քան բավական է, բայց երբեմն մենք կարիք ենք ունենում օգտագործելու իսկապես մեծ թվեր, օրինակ՝ գախտնագրման կամ 
```
```suggestion
Հիմնական դեպքերում սա ավելի քան բավական է, բայց երբեմն մենք կարիք ենք ունենում օգտագործելու իսկապես մեծ թվեր, օրինակ՝ գաղտնագրման կամ 
```

bugron commented on 1-js/02-first-steps/05-types/article.md:
```diff
@@ -1,226 +1,250 @@
 
-For most purposes that's quite enough, but sometimes we need really big numbers, e.g. for cryptography or microsecond-precision timestamps.
+Հիմնական դեպքերում սա ավելին քան բավական է, բայց երբեմն մենք կարիք ենք ունենում օգտագործելու իսկապես մեծ թվեր, օրինակ՝ գախտնագրման կամ 
+ժամանականիշի (timestamp) միկրովարկյաններով ներկայացման դեպքում։
```
No new line

bugron commented on 1-js/02-first-steps/05-types/article.md:
```diff
@@ -1,226 +1,250 @@
+ժամանականիշի (timestamp) միկրովարկյաններով ներկայացման դեպքում։
 
-`BigInt` type was recently added to the language to represent integers of arbitrary length.
+`BigInt`-ը վերջերս ավելացվել է լեզվում՝ կամայական մեծության թվեր նեկայացնելու նպատակով։
```
```suggestion
`BigInt`-ը վերջերս է ավելացվել լեզվում՝ կամայական մեծության թվեր նեկայացնելու նպատակով։
```

bugron commented on 1-js/02-first-steps/05-types/article.md:
````diff
@@ -1,226 +1,250 @@
 ```
 
-As `BigInt` numbers are rarely needed, we don't cover them here, but devoted them a separate chapter <info:bigint>. Read it when you need such big numbers.
+Քանի որ `BigInt` թվերը հազվադեպ են օգտագործվում, դրանց այստեղ չենք անրադառնա, այլ կդիտարկենք առանձին՝ <info:bigint> գլխում։ Կարդացե՛ք այն,
````
```suggestion
Քանի որ `BigInt` թվերը հազվադեպ են օգտագործվում, դրանց այստեղ չենք անդրադառնա, այլ կդիտարկենք առանձին՝ <info:bigint> գլխում։ Կարդացե՛ք այն,
```

bugron commented on 1-js/02-first-steps/05-types/article.md:
```diff
@@ -1,226 +1,250 @@
 
-As `BigInt` numbers are rarely needed, we don't cover them here, but devoted them a separate chapter <info:bigint>. Read it when you need such big numbers.
+Քանի որ `BigInt` թվերը հազվադեպ են օգտագործվում, դրանց այստեղ չենք անրադառնա, այլ կդիտարկենք առանձին՝ <info:bigint> գլխում։ Կարդացե՛ք այն,
+եթե նման մեծ թվերի օգտագոծման կարիք կունենաք։
```
No new line

bugron commented on 1-js/02-first-steps/05-types/article.md:
```diff
@@ -1,226 +1,250 @@
-You can check [*MDN* BigInt compatibility table](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt#Browser_compatibility) to know which versions of a browser are supported.
+Կարող եք ստուգել [*MDN* BigInt compatibility table](https://developer.mozilla.
+org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt#Browser_compatibility) հոդվածը, հասկանալու համար, թե զննիչների որ 
+տարբերակների հետ է այն համատեղելի։
```
Excessive new lines here.  Please, also review other places.

bugron commented on 1-js/02-first-steps/05-types/article.md:
```diff
@@ -1,226 +1,250 @@
+3. Թեք չակերտներ․ <code>&#96;Բարև&#96;</code>։
 
-Double and single quotes are "simple" quotes. There's practically no difference between them in JavaScript.
+Կրկնակի և միակի չակերտները «պարզ» չակերտներ են։ JavaScript-ում դրանց միջև տարբերություններ չկան։
```
```suggestion
Կրկնակի և եզակի չակերտները «պարզ» չակերտներ են։ JavaScript-ում դրանց միջև տարբերություններ չկան։
```
Let's keep it consistent.

bugron commented on 1-js/02-first-steps/05-types/article.md:
````diff
@@ -1,226 +1,250 @@
 ```
 
-The expression inside `${…}` is evaluated and the result becomes a part of the string. We can put anything in there: a variable like `name` or an arithmetical expression like `1 + 2` or something more complex.
+`${…}`-ի մեջ ներդրված արտահայտությունը գնահատվում է, և արդյունքը դառնում է տողի մի մասը։ Մենք կարող ենք տեղադրել այնեղ ամեն ինչ՝ 
````
```suggestion
`${…}`-ի մեջ ներդրված արտահայտությունը գնահատվում է, և արդյունքը դառնում է տողի մի մասը։ Մենք կարող ենք տեղադրել այնտեղ ամեն ինչ՝ 
```

bugron commented on 1-js/02-first-steps/05-types/article.md:
```diff
@@ -1,226 +1,250 @@
+`${…}`-ի մեջ ներդրված արտահայտությունը գնահատվում է, և արդյունքը դառնում է տողի մի մասը։ Մենք կարող ենք տեղադրել այնեղ ամեն ինչ՝ 
+փոփոխական (`name`) կամ թվաբանական արտահայտություն՝ `1 + 2`-ի նման կամ ավելի բարդ։
+
+Ուշադրություն դարձրեք այն փաստին, որ սա հնարավոր է միյայն թեք չակերտների դեպքում։ Մյուս չակերտները չունեն այսպիսի ներդրման 
```
```suggestion
Ուշադրություն դարձրեք այն փաստին, որ սա հնարավոր է միայն թեք չակերտների դեպքում։ Մյուս չակերտները չունեն այսպիսի ներդրման 
```

bugron commented on 1-js/02-first-steps/05-types/article.md:
````diff
@@ -1,226 +1,250 @@
 ```
 
-We'll cover strings more thoroughly in the chapter <info:string>.
+Մենք կանրադառնանք տողերին ավելի խորացված կերպեվ <info:string> գլխում։
````
```suggestion
Մենք կանրադառնանք տողերին ավելի մանրամասն <info:string> գլխում։
```

bugron commented on 1-js/02-first-steps/05-types/article.md:
````diff
@@ -1,226 +1,250 @@
 ```
 
-## Boolean (logical type)
+## Բուլյան (տրամաբանական տիպ)
````
```suggestion
## Boolean (տրամաբանական տիպ)
```
Here and after the term `boolean` should not be translated and should be used as it.

bugron commented on 1-js/02-first-steps/05-types/article.md:
````diff
@@ -1,226 +1,250 @@
 ```
 
-In JavaScript, `null` is not a "reference to a non-existing object" or a "null pointer" like in some other languages.
+JavaScript-ում `null`-ը «գոյություն չունեցող օբեկտի հղում» կամ «զրոյական ցուցիչ» չէ, ինչպես որոշ այլ լեզուներում։
````
```suggestion
JavaScript-ում `null`-ը «գոյություն չունեցող օբյեկտի հղում» կամ «զրոյական ցուցիչ» չէ, ինչպես որոշ այլ լեզուներում։
```

bugron commented on 1-js/02-first-steps/05-types/article.md:
```diff
@@ -1,226 +1,250 @@
 
-The `typeof` operator returns the type of the argument. It's useful when we want to process values of different types differently or just want to do a quick check.
+`typeof` օպերատորը վերադարձնում է իր արգումենտի տիպը։ Այն օգտակար է, երբ մենք ուզում ենք մշակել տարբեր տիպերի արժեքները տարբեր կերպով, կամ 
+պարզեպես արագ ստուգման կարիք կա։
```
```suggestion
պարզապես արագ ստուգման կարիք կա։
```

bugron commented on 1-js/02-first-steps/05-types/article.md:
```diff
@@ -1,226 +1,250 @@
+պարզեպես արագ ստուգման կարիք կա։
 
-It supports two forms of syntax:
+Այն ունի օգտագործման երկու սինտաքս․
```
Instead of the word `սինտաքս` I'd prefer to actually use a translation. I usually use Microsoft Terminology Translator in case I am in doubt for guidance. https://www.microsoft.com/en-us/language/Search?&searchTerm=syntax&langID=49&Source=true&productid=0

bugron replied:
In case I feel that the translated term feels too alien I also supply the original English word in parentheses for reference and then just use the translated term afterwards.

bugron commented on 1-js/02-first-steps/05-types/article.md:
```diff
@@ -248,29 +272,34 @@ typeof alert // "function"  (3)
-1. `Math` is a built-in object that provides mathematical operations. We will learn it in the chapter <info:number>. Here, it serves just as an example of an object.
-2. The result of `typeof null` is `"object"`. That's an officially recognized error in `typeof` behavior, coming from the early days of JavaScript and kept for compatibility. Definitely, `null` is not an object. It is a special value with a separate type of its own.
-3. The result of `typeof alert` is `"function"`, because `alert` is a function. We'll study functions in the next chapters where we'll also see that there's no special "function" type in JavaScript. Functions belong to the object type. But `typeof` treats them differently, returning `"function"`. That also comes from the early days of JavaScript. Technically, such behavior isn't correct, but can  ...
+1. `Math`-ը ներկառուցված օբեկտ է, որը տրամադրում է մաթեմատիկական տարբեր գործողություններ և հաստատուններ։ Մենք կուսումնասիրենք այն 
```
```suggestion
1. `Math`-ը ներկառուցված օբյեկտ է, որը տրամադրում է մաթեմատիկական տարբեր գործողություններ և հաստատուններ։ Մենք կուսումնասիրենք այն 
```

bugron commented on 1-js/02-first-steps/05-types/article.md:
```diff
@@ -248,29 +272,34 @@ typeof alert // "function"  (3)
+2. `typeof null`-ի արդյունքը `"object"` է։ Այն `typeof` օպերատորի վարքագծի պաշտոնապես ընդունված սխալ է, որը եկել է JavaScript-ի ստեղծման 
+   վաղ ժամանակներից և մնացել է լեզվում համատեղելիության նպատակով։ Անշուշտ, `null`-ը օբյեկտ չէ։ Այն հատուկ արժեք է, որը ունի իր սեփական տիպը։
+3. `typeof alert`-ի արդյունքը `"function"` է, քանի որ `alert`-ը ֆունկցիա է։ Մենք կուսումնասիրենք ֆունկցիաները հաջորդ գլուխներում, որտեղ 
+   կիմանանք, որ JavaScript-ը չունի առանձին «ֆունկցիա» տիպ։ Ֆունկցիաները պատկանում են օբյեկտ տիպին։ Բայց `typeof`-ը մշակում է դրանք 
```
՝«ֆունկցիա»՝ should not be translated here since it refers to the type `function` and not just the word function.

bugron commented on 1-js/02-first-steps/05-types/article.md:
```diff
@@ -248,29 +272,34 @@ typeof alert // "function"  (3)
-- `undefined` for unassigned values -- a standalone type that has a single value `undefined`.
-- `object` for more complex data structures.
-- `symbol` for unique identifiers.
+- `number` բոլոր տեսակի թվերի համար․ ամբողջ և լողացող կետով, ամբողջ թվերը սահմանափակված են <code>±(2<sup>53</sup>-1)</code> միջակայքում։
```
```suggestion
- `number` բոլոր տեսակի թվերի համար․ ամբողջ և տասնորդական, ամբողջ թվերը սահմանափակված են <code>±(2<sup>53</sup>-1)</code> միջակայքում։
```

bugron commented on 1-js/02-first-steps/06-alert-prompt-confirm/article.md:
```diff
@@ -1,105 +1,112 @@
 ## alert
 
-This one we've seen already. It shows a message and waits for the user to press "OK".
+Այս ֆունկցիայի հետ մենք արդեն առընչվել ենք։ Այն ցույց է տալիս հաղորդագրություն և սպասում է օգտատիրոջ՝ «OK» կոճակը սեղմելուն։
```
```suggestion
Այս ֆունկցիայի հետ մենք արդեն առնչվել ենք։ Այն ցույց է տալիս հաղորդագրություն և սպասում է օգտատիրոջ՝ «OK» կոճակը սեղմելուն։
```

bugron commented on 1-js/02-first-steps/06-alert-prompt-confirm/article.md:
````diff
@@ -1,105 +1,112 @@
 
-```smart header="The square brackets in syntax `[...]`"
-The square brackets around `default` in the syntax above denote that the parameter is optional, not required.
+```smart header="Քառակուսի փակագծերը [...] սինտաքսում"
````
```suggestion
```smart header="Քառակուսի փակագծերը `[...]` սինտաքսում"
```

bugron commented on 1-js/02-first-steps/06-alert-prompt-confirm/article.md:
````diff
@@ -1,105 +1,112 @@
 ```
 
-The visitor can type something in the prompt input field and press OK. Then we get that text in the `result`. Or they can cancel the input by pressing Cancel or hitting the `key:Esc` key, then we get `null` as the `result`.
+Օգտատերը կարող է մուտքագրման դաշտում լրացնել ինչ որ բան և սեղմել OK։ Մուտքագրված տեքստը կվերագրվի `result` փոփոխականին։ Օգտատերը նաև 
````
```suggestion
Օգտատերը կարող է մուտքագրման դաշտում լրացնել ինչ-որ բան և սեղմել OK։ Մուտքագրված տեքստը կվերագրվի `result` փոփոխականին։ Օգտատերը նաև 
```

bugron commented on 1-js/02-first-steps/06-alert-prompt-confirm/article.md:
`````diff
@@ -1,105 +1,112 @@
 
-````warn header="In IE: always supply a `default`"
-The second parameter is optional, but if we don't supply it, Internet Explorer will insert the text `"undefined"` into the prompt.
+````warn header="IE-ի համար միշտ սահմանեք լռելյայն (default) արժեք"
`````
```suggestion
````warn header="IE-ի համար միշտ սահմանեք լռելյայն (`default`) արժեք"
```

bugron commented on 1-js/02-first-steps/06-alert-prompt-confirm/article.md:
`````diff
@@ -1,105 +1,112 @@
 
-```js run
-let test = prompt("Test");
+````js run
`````
```suggestion
```js run
```

bugron reviewed (changes requested):
@Arthur-Gyulabyan thanks again for the PR. I am amazed by the quality of the translation, really!
Please do not get discouraged by the number of comments! Except for those new line comments there are just a few things to address and we'll be ready to merge this.
Way to go!

arthur-gyulabyan committed:
fix: unnecessary new lines removed, typos fixed

arthur-gyulabyan committed:
fix: unnecessary new lines removed, typos fixed

arthur-gyulabyan commented:
@bugron thank you for review! The issue with new lines occurred because of my editor's settings. I fixed it, so now everything is okay. I also fixed typos and other bugs.

arthur-gyulabyan commented:
/done

bugron reviewed (approved):
LGTM

bugron commented:
Thanks for your first PR! Merging and looking forward to your future contributions. 🎉🎉🎉
